nimble/host/mesh/src/friend.c

```diff
@@ -1224,7 +1224,6 @@ static void friend_timeout(struct ble_npl_event *work)
 	}
 
 	/* Clear the flag we use for segment tracking */
-	BT_MESH_ADV(frnd->last)->flags &= ~NET_BUF_FRAGS;
 	BT_MESH_ADV(frnd->last)->flags = 0;
```

Comment:
This is a typo. The second line should be `frags` and you should set it to NULL (this is how it looks in Zephyr). And we should not erase the first line. We need to double-check this.

Reply:
I checked this method with Zephyr and that typo and value is the only difference. Synced with Zephyr

Reply:
As disgussed offline, this requires bigger fix. I'll remove this from this PR